ldap-client.go

```diff
@@ -254,6 +256,50 @@ func (lc *LDAPClient) GetAllGroupsByName(groupName string) ([]LdapGroup, error)
 	return groups, nil
 }
 
+// GetAllGroupsWithMembersByName returns list of groups matching a name.
+func (lc *LDAPClient) GetAllGroupsWithMembersByName(groupsDN string, groupCN []string) ([]*LdapGroup, error) {
+	err := lc.Connect()
+	if err != nil {
+		return nil, err
+	}
+
+	dn := lc.BindDN
+	if groupsDN != "" {
+		dn = groupsDN
+	}
+	// First bind with a read only user
+	if dn != "" && lc.BindPassword != "" {
```

Comment:
Don't we wish to return an error here?
Because we can't continue if there's no baseDN and password to bind to.

Reply:
search will generate us some good error